### PR TITLE
add graphql auth middleware

### PR DESCRIPTION
**Pull request by atomboulian**
Helpful links for context: https://github.com/apollographql/apollo-link/tree/master/packages/apollo-link-http#upgrade-apollo-client-10

### CODE REVIEW

File: src/components/auth/middleware.js [hunk 1]
```diff
@@ -84,7 +87,7 @@ const formatMeta = when(
  * @func
  * @sig a -> {k: v} -> ({k: v} -> {k: v}) -> {k: v} -> undefined
  */
-export default service => () => next => action => {
+export const serviceAuthMiddleware = service => () => next => action => {
```
Comment (atomboulian) [inline src/components/auth/middleware.js]: Will need to update other libs referencing this to use `import {serviceAuthMiddleware} from attainia-web-components/auth/middleware` of course but according to the style guide we prefer non defaults where possible. And, we're adding a middleware export for the auth for GraphQL to this file.

**Comment by TimGraf** [comment]
LGTM

**Comment by AlexMaskovyak** [comment]
LGTM

File: src/components/auth/middleware.js [hunk 2]
```diff
@@ -93,3 +96,32 @@ export default service => () => next => action => {
...
+/**
+ * Sets up the apollo client link with auth headers.
+ *
+ * Assume an apollo client with an apollo-http-link is passed in.
+ * https://www.apollographql.com/docs/link/links/http.html
+ *
+ * In our link composition, we create a middleware link and place it
+ * before calling any other links. Eventually the last link typically makes
+ * the network call.
+ * https://www.apollographql.com/docs/link/overview.html
```
Comment (arizonatribe) [inline src/components/auth/middleware.js]: Good information to provide here!

File: src/components/auth/middleware.js [hunk 3]
```diff
@@ -93,3 +96,32 @@ export default service => () => next => action => {
...
+ *
+ * @func
+ * @sig a -> {k: v} -> ({k: v} -> {k: v}) -> {k: v} -> undefined
+ */
+export const apolloAuthMiddleWare = appolloClientLink => next => action => {
+    if ([PARSED_TOKEN, LOGIN].includes(action.type)) {
+        const middlewareLink = new ApolloLink((operation, forward) => {
```
Comment (arizonatribe) [inline src/components/auth/middleware.js]: Definitely an interesting new API for Apollo.

**Comment by arizonatribe** [comment]
LGTM2

**Pull request merged**